src/sentry/api/fields/sentry_slug.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+from django.utils.translation import gettext_lazy as _
+from drf_spectacular.types import OpenApiTypes
+from drf_spectacular.utils import extend_schema_field
+from rest_framework import serializers
+
+# (?![0-9]+$) - Negative lookahead to ensure the slug is not entirely numeric
+# [a-z0-9_\-] - Matches lowercase letters, numbers, underscores, and hyphens
+MIXED_SLUG_PATTERN = r"^(?![0-9]+$)[a-z0-9_\-]+$"
+DEFAULT_SLUG_ERROR_MESSAGE = _(
+    "Enter a valid slug consisting of lowercase letters, numbers, underscores or hyphens. "
+    "It cannot be entirely numeric."
+)
+
+
+@extend_schema_field(field=OpenApiTypes.STR)
+class SentrySlugField(serializers.RegexField):
+    """
+    A regex field which validates that the input is a valid slug. Allowed
+    characters are lowercase letters, numbers, underscores, and hyphens. The
+    slug cannot be entirely numeric.
+    """
+
+    default_error_messages = {
+        "invalid": DEFAULT_SLUG_ERROR_MESSAGE,
+    }
+
+    def __init__(
+        self,
+        regex=MIXED_SLUG_PATTERN,
+        error_messages=None,
+        *args,
+        **kwargs,
+    ):
+        # Avoid using mutable dict for error_messages defaults b/c all calls to
+        # the function reuse this one instance, persisting changes between them.
+        if error_messages is None:
+            error_messages = self.default_error_messages
```

Comment:
could you do `dict.copy()`?

Reply:
yeah sure

Reply:
oh i meant instead of `error_messages = None` in `__init__`, would it work to initialize it to a copy of the dict?

Reply:
or initialize it to just `{"invalid": DEFAULT_SLUG_ERROR_MESSAGE}` so it creates a new dict each time?

Reply:
It warns for adding function calls in arguments saying:
`Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.Flake8(B008)`

You can't do `error_messages={"invalid": DEFAULT_SLUG_ERROR_MESSAGE}` cuz then it says:
`Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.Flake8(B006)`

Reply:
huh interesting. i guess ignore what i said then!